src/native/corehost/apphost/apphost.windows.cpp

```diff
@@ -86,7 +89,52 @@ namespace
         return msg;
     }
 
-    void show_error_dialog(const pal::char_t *executable_name, int error_code)
+    class activation_context final
+    {
+    public:
+        activation_context(const pal::char_t* source, const pal::char_t* resource_name)
+            : m_activated{false}
+            , m_context_handle{INVALID_HANDLE_VALUE}
+            , m_cookie{}
+        {
+            ACTCTXW actctx = { sizeof(ACTCTXW), 0, source };
+            if (resource_name != nullptr)
```

Comment:
Going with the idea of minimal functionality - this is always going to be non-null.

Reply:
we could use a dev/debug-only assert here